File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    //user parameters we can derive from the pagination params class
    public class UserParams : PaginationParams
    {

        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        //To filter the user to be min of 18 yrs and max of 100 yrs
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 100;
        public string OrderBy { get; set; } = "lastActive";
    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    //reduce the page size and we'll just set this to three so we can more easily see how this
    pageSize = 5;
    orderBy = 'lastActive';

    //If the logged in user is female, set this gender to be male
    //this means that if the user's gender is female, then we're going to request male users
    constructor(user: User) {
        this.gender  = user.gender === 'female' ? 'male' : 'female'
    }
}